### PR TITLE
chore(doc): enrich CONTRIBUTING.md with all the necessary steps

### PR DESCRIPTION
**Pull request by dubloom**

### CODE REVIEW

**Review by pablomartinezbernardo — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -40,13 +40,27 @@ options:
...
+## Compiling
+
+### Build httpd
+
+In order to build the module you have to build httpd first:
+```sh
+cd httpd 
+./configure --with-included-apr --prefix=$(pwd)/httpd-build --enable-mpms-shared="all" 
+make -j8 
+make install
```
Comment (dmehala) [inline CONTRIBUTING.md:52]: Building and installing httpd is not required.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -40,13 +40,27 @@ options:
                         Directory where HTTPD will be extracted
 ````
 
-### Compiling
+## Compiling
+
+### Build httpd
+
+In order to build the module you have to build httpd first:
```
Comment (dmehala) [inline CONTRIBUTING.md:47]: That is inaccurate. Only configuring the `httpd` is required.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -40,13 +40,27 @@ options:
...
+
+### Build the module
+
 CMake is our build system. If you are not familiar with CMake, read [the tutorial.](https://cmake.org/cmake/help/latest/guide/tutorial/index.html)
 
 Configure and compile all targets in release:
 
 ````sh
-cmake -B build -DCMAKE_BUILD_TYPE=Release -DHTTPD_SRC_DIR=httpd-2.4 .
+git submodule update --init --recursive
```
Comment (dmehala) [inline CONTRIBUTING.md:62]: May I ask to move it in a separate section instructing how to clone the repo?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -40,13 +40,27 @@ options:
                         Directory where HTTPD will be extracted
 ````
 
-### Compiling
+## Compiling
+
+### Build httpd
```
Comment (dmehala) [inline CONTRIBUTING.md:45]:
```suggestion
### Setup `httpd`
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -40,13 +40,27 @@ options:
...
+
+In order to build the module you have to build httpd first:
+```sh
+cd httpd 
+./configure --with-included-apr --prefix=$(pwd)/httpd-build --enable-mpms-shared="all" 
+make -j8 
+make install
+```
```
Comment (dmehala) [inline CONTRIBUTING.md:53]:
```suggestion
```sh
python scripts/setup-httpd.py -o httpd
cd httpd
./configure --with-included-apr --prefix=$(pwd)/httpd-build --enable-mpms-shared="all" 
```
```

**Review by dmehala — COMMENTED** [review]
Good addition. Added few comments.

**Review by dmehala — APPROVED** [review]

**Pull request merged**